SimPEG/regularization/base.py

```diff
@@ -560,33 +1044,174 @@ def _cell_distances(self):
 
     @property
     def orientation(self):
+        """Direction along which smoothness is enforced.
+
+        Returns
+        -------
+        str
+            The direction along which smoothness is enforced. On of {'x','y','z'}
+
+        """
         return self._orientation
 
 
 class SmoothnessSecondOrder(SmoothnessFirstOrder):
-    """
-    This base class regularizes on the second
-    spatial derivative, optionally normalized by the base cell size.
-
-    **Optional Inputs**
-
-    :param discretize.base.BaseMesh mesh: SimPEG mesh
-    :param int nP: number of parameters
-    :param IdentityMap mapping: regularization mapping, takes the model from
-        model space to the space you want to regularize in
-    :param numpy.ndarray reference_model: reference model
-    :param numpy.ndarray active_cells: active cell indices for reducing the
-        size of differential operators in the definition of a regularization mesh
-    :param numpy.ndarray weights: cell weights
-    :param bool reference_model_in_smooth: include the reference model in the
-        smoothness computation? (eg. look at Deriv of m (False) or Deriv of
-        (m-reference_model) (True))
-    :param numpy.ndarray weights: vector of cell weights (applied in all terms)
+    r"""Second-order smoothness (flatness) least-squares regularization.
+
+    ``SmoothnessSecondOrder`` regularization is used to ensure that values in the recovered
+    model have small second-order spatial derivatives. When a reference model is included,
+    the regularization preserves second-order smoothness within the reference model along
+    the direction specified (x, y or z). Optionally, custom cell weights can be used
+    to control the degree of smoothness being enforced throughout different regions
+    the model.
+
+    See the *Notes* section below for a comprehensive description.
+
+    Parameters
+    ----------
+    mesh : discretize.base.BaseMesh mesh
+        The mesh on which the regularization is discretized.
+    orientation : {'x', 'y', 'z'}
+        The direction along which smoothness is enforced.
+    active_cells : None, (n_cells, ) numpy.ndarray of bool
+        Boolean array defining the set of :py:class:`~.regularization.RegularizationMesh`
+        cells that are active in the inversion. If ``None``, all cells are active.
+    mapping : None, SimPEG.maps.BaseMap
+        The mapping from the model parameters to the active cells in the inversion.
+        If ``None``, the mapping is the identity map.
+    reference_model : None, (n_param, ) numpy.ndarray
+        Reference model. If ``None``, the reference model in the inversion is set to
+        the starting model. To include the reference model in the regularization, the
+        `reference_model_in_smooth` property must be set to ``True``.
+    reference_model_in_smooth : bool, optional
+        Whether to include the reference model in the smoothness terms.
```

Comment:
Same reasoning as before.
```suggestion
        Whether to include the reference model in the smoothness term.
```

Reply:
Fixed